Flaky test test_tracking[legacy_poly_experiment_cancelled_by_max_runtime]

**Describe the bug**
```
            for (
                iter_expression,
                snapshot_expression,
                expected,
            ) in assert_present_in_snapshot:
                for i, snapshot in snapshots.items():
                    if re.match(iter_expression, str(i)):
>                       check_expression(
                            snapshot.to_dict(),
                            snapshot_expression,
                            expected,
                            f"Snapshot {i} did not match:\n",
                        )

/Users/runner/work/ert/ert/tests/ert_tests/status/test_tracking_integration.py:265: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

original = {'metadata': {'iter': 0}, 'reals': {'0': {'active': True, 'end_time': None, 'start_time': None, 'status': 'Failed', ...}, '1': {'active': True, 'end_time': None, 'start_time': None, 'status': 'Failed', ...}}, 'status': 'Starting'}
path_expression = 'reals.*.steps.*.jobs.*.status', expected = 'Failed'
msg_start = 'Snapshot 0 did not match:\n'

    def check_expression(original, path_expression, expected, msg_start):
        assert isinstance(original, dict), f"{msg_start}data is not a dict"
        jsonpath_expr = parse(path_expression)
        match_found = False
        for match in jsonpath_expr.find(original):
            match_found = True
>           assert (
                match.value == expected
            ), f"{msg_start}{str(match.full_path)} value ({match.value}) is not equal to ({expected})"
E           AssertionError: Snapshot 0 did not match:
E             reals.0.steps.0.jobs.0.status value (Running) is not equal to (Failed)
E           assert 'Running' == 'Failed'
E             - Failed
E             + Running

/Users/runner/work/ert/ert/tests/ert_tests/status/test_tracking_integration.py:34: AssertionError
------------------------------ Captured log call -------------------------------
ERROR    res.job_queue.job_queue_node:job_queue_node.py:179 MAX_RUNTIME reached in run path /private/var/folders/24/8k48jl6d249_n_qfxwsl6xvm0000gn/T/pytest-of-runner/pytest-0/test_tracking_legacy_poly_expe0/max_runtime_poly_example/max_runtime_poly_example/poly_out/real_0/iter_0
ERROR    res.job_queue.job_queue_node:job_queue_node.py:179 MAX_RUNTIME reached in run path /private/var/folders/24/8k48jl6d249_n_qfxwsl6xvm0000gn/T/pytest-of-runner/pytest-0/test_tracking_legacy_poly_expe0/max_runtime_poly_example/max_runtime_poly_example/poly_out/real_1/iter_0
```

**To Reproduce**
Happens quite often, but seems you need python3.8 and macOS

--- a/ert_shared/models/base_run_model.py
+++ b/ert_shared/models/base_run_model.py
@@ -102,13 +102,16 @@ class BaseRunModel(object):
             self._fail_message = str(e)
             self._simulationEnded()
 
-        self._run_context = None  # delete last active run_context to notify fs_manager that storage is not being written to
-
     def runSimulations(self, job_queue, run_context):
         raise NotImplementedError("Method must be implemented by inheritors!")
 
     def create_context(self, arguments):
         raise NotImplementedError("Method must be implemented by inheritors!")
+
+    def teardown_context(self):
+        # Used particularly to delete last active run_context to notify
+        # fs_manager that storage is not being written to.
+        self._run_context = None
 
     @job_queue(None)
     def killAllSimulations(self):
@@ -239,25 +242,34 @@ class BaseRunModel(object):
             timed_out = self._job_queue.did_job_time_out(queue_index)
 
         # Avoids reading from disk for jobs in these states since there's no
-        # data anyway
-        if status in [
-            JobStatusType.JOB_QUEUE_PENDING,
-            JobStatusType.JOB_QUEUE_SUBMITTED,
-            JobStatusType.JOB_QUEUE_WAITING,
-        ]:
+        # data anyway. If timed out, never exit here as that would prevent
+        # propagation of the failure status.
+        if (
+            status
+            in [
+                JobStatusType.JOB_QUEUE_PENDING,
+                JobStatusType.JOB_QUEUE_SUBMITTED,
+                JobStatusType.JOB_QUEUE_WAITING,
+            ]
+            and not timed_out
+        ):
             return
 
         fms = self.realization_progress[iteration].get(run_arg.iens, None)
+        jobs = fms[0] if fms else None
 
         # Don't load from file if you are finished
-        if fms and BaseRunModel.is_forward_model_finished(fms[0]):
-            jobs = self.realization_progress[iteration][run_arg.iens][0]
-        else:
-            fms = ForwardModelStatus.load(run_arg.runpath, num_retry=1)
-            if not fms:
+        if not fms or not BaseRunModel.is_forward_model_finished(fms[0]):
+            loaded = ForwardModelStatus.load(run_arg.runpath, num_retry=1)
+            if not loaded and not timed_out:
+                # If this idx timed out, returning here would prevent
+                # non-successful jobs in being marked as failed (timed out). So
+                # return only in the case where it did not time out.
                 return
 
-            jobs = fms.jobs
+            if loaded:
+                jobs = loaded.jobs
+
         if timed_out:
             for job in jobs:
                 if job.status != "Success":

--- a/ert_shared/status/tracker/evaluator.py
+++ b/ert_shared/status/tracker/evaluator.py
@@ -131,6 +131,7 @@ class EvaluatorTracker:
         self._work_queue.put(EvaluatorTracker.DONE)
         self._work_queue.join()
         drainer_logger.debug("tasks complete")
+        self._model.teardown_context()
 
     def track(self):
         while True:

--- a/ert_shared/status/tracker/legacy.py
+++ b/ert_shared/status/tracker/legacy.py
@@ -112,6 +112,8 @@ class LegacyTracker:
         yield from self._retroactive_update_event()
         yield self._end_event()
 
+        self._model.teardown_context()
+
     def _create_snapshot_dict(
         self,
         run_context: ErtRunContext,

